src/quoting/spline_pool.rs

```diff
@@ -0,0 +1,487 @@
+use crate::math::uint::U256;
+use crate::quoting::base_pool::{
+    BasePool, BasePoolResources, BasePoolState, BasePoolQuoteError,
+    MAX_TICK_AT_MAX_TICK_SPACING, MIN_TICK_AT_MAX_TICK_SPACING, MAX_TICK_SPACING,
+};
+use crate::quoting::types::{BlockTimestamp, NodeKey, Pool, Quote, QuoteParams, Tick};
+use alloc::vec;
+use core::ops::Add;
+use num_traits::{ToPrimitive};
+
+#[derive(Clone, Copy)]
+pub struct SplinePoolState {
+    pub base_pool_state: BasePoolState,
+    pub liquidity_factor: u128,
+    pub total_shares: u256,
+}
+
+#[derive(Default, Clone, Copy)]
+pub struct SplinePoolResources {
+    pub base_pool_resources: BasePoolResources,
+    pub fee_compounds: u32,
+}
+
+impl Add for SplinePoolResources {
+    type Output = Self;
+    fn add(self, rhs: Self) -> Self {
+        Self {
+            base_pool_resources: self.base_pool_resources + rhs.base_pool_resources,
+            fee_compounds: self.fee_compounds + rhs.fee_compounds,
+        }
+    }
+}
+
+pub struct SplinePool {
+    base_pool: BasePool,
+    liquidity_factor: u128,
+    total_shares: u256,
+    last_compound_time: u64,
+}
+
+impl SplinePool {
+    pub fn new(
+        token0: U256,
+        token1: U256,
+        extension: U256,
+        sqrt_ratio: U256,
+        liquidity_factor: u128,
+        last_compound_time: u64,
+    ) -> Self {
+        let signed_liquidity: i128 =
+            liquidity_factor.to_i128().expect("liquidity_factor exceeds i128");
+
+        let ticks = if liquidity_factor > 0 {
+            vec![
+                Tick {
+                    index: MIN_TICK_AT_MAX_TICK_SPACING,
+                    liquidity_delta: signed_liquidity,
+                },
+                Tick {
+                    index: MAX_TICK_AT_MAX_TICK_SPACING,
+                    liquidity_delta: -signed_liquidity,
+                },
+            ]
+        } else {
+            vec![] // Empty ticks for zero liquidity
+        };
+
+        SplinePool {
+            base_pool: BasePool::new(
+                NodeKey {
+                    token0,
+                    token1,
+                    fee: 0,
```

Comment:
don't think it should be `fee: 0` since they won't quote swaps properly given pools have nonzero fee.

would instead have SplinePool::new use similar input param structure to base pool, but using `SplinePoolState`

```rust
pub fn new(key: NodeKey, state: SplinePoolState, sorted_ticks: Vec<Tick>) -> Self
```

that way can feed in NodeKey into `BasePool::new`